src/kdc/kdc_preauth.c

```diff
@@ -1300,26 +1299,18 @@ return_padata(krb5_context context, krb5_kdcpreauth_rock rock,
             size++;
     }
 
-    if ((send_pa_list = malloc((size+1) * sizeof(krb5_pa_data *))) == NULL)
-        return ENOMEM;
-    if ((pa_order = malloc((size+1) * sizeof(int))) == NULL) {
-        free(send_pa_list);
-        return ENOMEM;
-    }
+    pa_order = k5calloc(size + 1, sizeof(int), &retval);
+    if (pa_order == NULL)
+        goto cleanup;
```

Comment:
`original_key` hasn't been initialized here.

Reply:
Good catch.  I added a memset at the start, and checked that the other cleaned up variables are initialized.